### ISSUE TITLE
limiting hunter_add_package architecture list for faster rebuild-all

### ISSUE THREAD

**Issue by headupinclouds** [ruslo/hunter#84]
A default hunter_add_package call like this:
```
hunter_add_package(OpenCV)
find_package(OpenCV REQUIRED)
```
in a cmake/polly build for ios-8-2
```
build.py --install ios-8-2 
```
seems to build : 386 x86_64 armv7 armv7s arm64 
```
lipo -info ${HUNTER_ROOT}/hunter/_Base/xxxxxxx/7bc3715/9ffc177/Install/lib/libopencv_flannd.a
Architectures in the fat file: /Users/dhirvonen/devel/hunter/_Base/xxxxxxx/7bc3715/9ffc177/Install/lib/libopencv_flannd.a are: i386 x86_64 armv7 armv7s arm64 
```
for both Debug and Release configurations, for a total of 10 build passes of OpenCV.  During development, I'm typically interested in just one of these, say arm64 in Release mode.  I generally don't need Debug builds of external packages I'm using, and will be developing and testing on a single platform.  Are there any hooks to support this type of hunter package configuration/limitation (architecture and optimization level)? 

If not, I would submit this as a feature request.  It could be a useful workaround for long build times in common use cases, when small changes are made to cmake/Hunter/config.cmake that force a rebuild-all of all hunter packages.  Limiting development builds to a single architecture could make development and package tuning more feasible.

**Comment by ruslo** [comment]
Fast fix with two architectures `ios-8-2-i386-arm64` from branch `develop`:
* https://github.com/ruslo/polly/blob/develop/ios-8-2-i386-arm64.cmake

There are two problems I currently investigating:
* In fact for some reason Xcode rebuild one architecture so for time measurement it's like +1
* Current CMake patched version for iOS expect two libraries, so i386 + arm64 is minimal combination. I will fix it later

> I generally don't need Debug builds of external packages I'm using

this feature is designed to solve such problems: https://github.com/ruslo/hunter/issues/22

**Comment by headupinclouds** [comment]
Great, so to translate this into a build command, it looks like we need hunter v0.8.13 (for the parallel build hunter CMake macros) and the latest poly in develop (if using build.py front end for --jobs command)
```
HunterGate(
    URL "https://github.com/ruslo/hunter/archive/v0.8.13.tar.gz"
    SHA1 "7103f0ab481523ca93a06c1c55abcd56741d3a97"
    FILEPATH "${CMAKE_BINARY_DIR}/hunter-config.cmake"
)
```
Then we could achieve a relatively fast build using:
```
build.py --toolchain ios-8-2-i386-arm64 --jobs 8 --clear --verbose
```
I'm not clear on how to make use of the release only builds mentioned in #22.   

That issue mentions:

> Create one build scheme url_sha1_cmake.cmake.in (?) instead of two url_sha1_release_debug.cmake.in url_sha1_combined_release_debug.cmake.in.

Does this require adding a new hunter.cmake file for each target:

Where this:
```
# Pick a download scheme
hunter_pick_scheme(
    DEFAULT
    url_sha1_release_debug # For separate release/debug generators, like Makefile
    COMBINED
    url_sha1_combined_release_debug # For Xcode and Visual Studio
)
```
is replaced with something like this:
```
# Pick a download scheme
hunter_pick_scheme(
    DEFAULT
    url_sha1_cmake # For separate release/debug generators, like Makefile
)
```

**Comment by ruslo** [comment]
> if using build.py front end for --jobs command

Actually they are not related. 'build.py --jobs' is for your local build. Hunter will use number of logical cores by default, if you want to rewrite this settings you can set HUNTER_JOBS_NUMBER manually. I.e. if you want to build both local project and 3rd party using 8 cores you can run:
```
> build.py --jobs 8 --fwd HUNTER_JOBS_NUMBER=8
```

> That issue mentions

It's just an implementation notes

> Does this require adding a new hunter.cmake file for each target

Yes hunter.cmake for CMake based projects will use single url_sha1_cmake instead of two url_sha1_release_debug + url_sha1_combined_release_debug

> I'm not clear on how to make use of the release only builds mentioned in #22

You just need to set HUNTER_CONFIGURATION_TYPES=Release and all 3rd party libraries will be built in Release configuration.

**Comment by headupinclouds** [comment]
Posting your response for the benefit of others:

>  the hunter.cmake change of all files will be made by me (and only once). Later you can just add new option HUNTER_CONFIGURATION_TYPES to your project build: build.py --fwd HUNTER_CONFIGURATION_TYPES=Release This will effectively register new Toolchain-ID in which all 3rd party library will be built on Release mode by default (you still be able to customize it per package in config.cmake). alternatively you can set an option in your CMakeLists.txt: option(HUNTER_CONFIGURATION_TYPES Release)

**Comment by ruslo** [comment]
@headupinclouds Hi, David. I've added toolchain `ios-8-2-arm64` with only one architecture: https://github.com/ruslo/polly/blob/master/ios-8-2-arm64.cmake but you need to update also new patched CMake version which knows how to skip universal build for such kind of toolchain: https://github.com/ruslo/CMake/releases/tag/v3.2.1-p3
This version of Hunter [v0.9.0](https://github.com/ruslo/hunter/releases/tag/v0.9.0) learn new option [HUNTER_CONFIGURATION_TYPES](https://github.com/ruslo/hunter/wiki/usr.variables#hunter_configuration_types) which can be used to control Release/Debug/... builds of 3rd party packages. In short you need to run this command to build Release packages with only arm64 architecture:
```bash
> build.py --toolchain ios-8-2-arm64 --verbose --fwd HUNTER_CONFIGURATION_TYPES=Release
```

**Comment by headupinclouds** [comment]
Thanks, that should make a big difference in build times.  I've recompiled my CMake using v3.2.1-p3, and tested tested this with the Xcode (OS X) and iOS-8-2-arm64 toolchain.  The OS X Xcode build works:

```
 build.py --toolchain ios-8-2-arm64 --verbose --fwd HUNTER_CONFIGURATION_TYPES=Release --jobs 8 --open
```

But I seem to hit a problem in a boost library for the iOS build:

```
build.py --toolchain ios-8-2-arm64 --verbose --fwd HUNTER_CONFIGURATION_TYPES=Release --jobs 8 --open
```
This is the output:
```
-- [hunter *** DEBUG *** 2015-04-22T21:37:59] Scheme: url_sha1_boost_ios_library

[hunter ** INTERNAL **] Unexpected empty string
[hunter ** INTERNAL **] [Directory:/Users/dhirvonen/devel/hunter/_Base/xxxxxxx/cf868fd/7d97147/Build/Boost/filesystem]
```

The ios-8-2 toolchain behaves the same.

**Comment by ruslo** [comment]
> But I seem to hit a problem in a boost library for the iOS build

Your pull request and additional fixes applied in branch develop or in [v0.9.1](https://github.com/ruslo/hunter/releases/tag/v0.9.1)
